### PR TITLE
Port the summary frontend response struct to OSS

### PR DESCRIPTION
**Pull request by tony-yang**
Port the summary frontend response struct from internal to the OSS summarizer component. Updated all json fields that use underscore as the word separator to dash.

### CODE REVIEW

File: testgrid/cmd/summarizer/summary.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	OpenTestTemplate      linkTemplate `json:"open-test-template"`
+	FileBugTemplate       linkTemplate `json:"file-bug-template"`
+	AttachBugTemplate     linkTemplate `json:"attach-bug-template"`
+	ResultsURLTemplate    linkTemplate `json:"results-url-template"`
+	CodeSearchURLTemplate linkTemplate `json:"code-search-url-template"`
+	AboutDashboardURL     string       `json:"about-dashboard-url"`
+	OpenBugTemplate       linkTemplate `json:"open-bug-template"`
+
+	ResultsText string `json:"results-text"`
+	LatestGreen string `json:"latest-green"` // TODO(amerai): make this not always empty?
```
Comment (michelle192837) [inline testgrid/cmd/summarizer/summary.go]: Remove comment

File: testgrid/cmd/summarizer/summary.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	// sequences of values start. See graph_test.go for example.
+	Layout []int32
+
+	// Non-empty values for each test result.
+	Value []float64
+}
+
+type graph struct {
+	metric string
+	// TODO(amerai): make this []struct{name,label,values} instead.
```
Comment (michelle192837) [inline testgrid/cmd/summarizer/summary.go]: Remove comment

File: testgrid/cmd/summarizer/summary.go (hunk 1 above)
Comment (tony-yang) [reply testgrid/cmd/summarizer/summary.go]: Done

File: testgrid/cmd/summarizer/summary.go (hunk 2 above)
Comment (tony-yang) [reply testgrid/cmd/summarizer/summary.go]: Done

**Comment by Katharine** [comment]
> Updated all json fields that use underscore as the word separator to dash.

Out of curiosity: why? Underscores are the usual approach.

**Comment by tony-yang** [comment]
I have no preference using either. It was just that in Testgrid, most attributes (80 to 90%) are already using dash. Only a few used underscores for whatever reason. For convenience, I converted the underscore to dash. If we have a style guide that preferred underscore, I'm happy to apply the style guide. For now, it's only because there's fewer underscore to change.

**Comment by michelle192837** [comment]
Hmm, Katharine has a point actually. There is a Google style guide (https://g3doc.corp.google.com/company/teams/cloud-graphite/index.md?cl=head) and we should probably  follow that (plus that would make the naming conventions more similar between JavaScript and Go). It'll require an internal change though to get all the JSON names updated properly. :/

Can you just update the field names back to match exactly, and we can do a larger variable-name-overhaul after?

**Comment by tony-yang** [comment]
Sounds good, as long as we have a convention, happy to use it. I'll make the change right now.

**Comment by stevekuznetsov** [comment]
30000/10 excited for these :)

**Comment by michelle192837** [comment]
Also excited :D (Though don't count on them yet :P Format will be changing for a few reasons in the short-term)

Thanks Tony!

/lgtm
/approve

**Comment by tony-yang** [comment]
Thanks all for the excitement and for providing feedbacks.

**Pull request merged**